core/event.h

```diff
@@ -68,12 +68,18 @@ class event_loop
 class gasmix_loop {
 	const struct dive &dive;
 	const struct divecomputer &dc;
-	struct gasmix last;
+	bool first_run;
 	event_loop loop;
-	const struct event *ev;
+	const struct event *next_event;
+	int last_cylinder_index;
+	int last_time;
 public:
 	gasmix_loop(const struct dive &dive, const struct divecomputer &dc);
-	gasmix next(int time);
+	std::pair<int, int> next_cylinder_index(); // -1 -> end
```

Comment:
Would be great if you could add a comment, what these functions do and what their return values are.

Reply:
Done.